`@Observable` Macro does not support properties with the `package` access modifier

### Description

If a property with `package` access modifier exists in a class with `@Observable` Macro, the build will fail.

### Reproduction

#### ✅ OK

```swift
import Observation

@Observable
class MyObservable1 {
    var value = 0
}
```

#### ❌ Build failed

```swift
import Observation

@Observable
class MyObservable2 {
    package var value = 0
}
```

![](https://github.com/apple/swift/assets/13805382/6cf7978d-e9fd-4ff1-a46e-d84485975866)

```sh
% swift build

Building for debugging...
error: emit-module command failed with exit code 1 (use -v to see invocation)
/var/folders/78/4hltfk2n2fz2f88371bhlp_h0000gn/T/swift-generated-sources/@__swiftmacro_20ObservablePlayground13MyObservable2C5value18ObservationTrackedfMp_.swift:2:5: error: duplicate modifier
    package  var _value  = 0
    ^~~~~~~
/Users/treastrain/Developer/GitHub.com/treastrain/ObservablePlayground/Sources/main.swift:11:17: note: in expansion of macro 'ObservationTracked' here
    package var value = 0
                ^~~~~
/var/folders/78/4hltfk2n2fz2f88371bhlp_h0000gn/T/swift-generated-sources/@__swiftmacro_20ObservablePlayground13MyObservable2C5value18ObservationTrackedfMp_.swift:1:21: note: modifier already specified here
@ObservationIgnored private
                    ^~~~~~~
/Users/treastrain/Developer/GitHub.com/treastrain/ObservablePlayground/Sources/main.swift:11:17: note: in expansion of macro 'ObservationTracked' here
    package var value = 0
                ^~~~~
/var/folders/78/4hltfk2n2fz2f88371bhlp_h0000gn/T/swift-generated-sources/@__swiftmacro_20ObservablePlayground13MyObservable2C5value18ObservationTrackedfMp_.swift:2:5: error: duplicate modifier
    package  var _value  = 0
    ^~~~~~~
/Users/treastrain/Developer/GitHub.com/treastrain/ObservablePlayground/Sources/main.swift:11:17: note: in expansion of macro 'ObservationTracked' here
    package var value = 0
                ^~~~~
/var/folders/78/4hltfk2n2fz2f88371bhlp_h0000gn/T/swift-generated-sources/@__swiftmacro_20ObservablePlayground13MyObservable2C5value18ObservationTrackedfMp_.swift:1:21: note: modifier already specified here
@ObservationIgnored private
                    ^~~~~~~
/Users/treastrain/Developer/GitHub.com/treastrain/ObservablePlayground/Sources/main.swift:11:17: note: in expansion of macro 'ObservationTracked' here
    package var value = 0
                ^~~~~
error: fatalError
```

### Expected behavior

Expect the access modifier for properties with `@ObservationIgnored` generated by that macro to be `private` rather than the syntactically incorrect `package private`.

![](https://github.com/apple/swift/assets/13805382/6b91fc7a-69fc-4dbd-927d-6cc34dcbd9fa)

### Environment

```sh
% swiftc -version

swift-driver version: 1.87.3 Apple Swift version 5.9.2 (swiftlang-5.9.2.2.56 clang-1500.1.0.2.5)
Target: arm64-apple-macosx14.0
```

### Additional information

- https://x.com/treastrain/status/1749161300213760344 (Japanese)

--- a/lib/Macros/Sources/ObservationMacros/ObservableMacro.swift
+++ b/lib/Macros/Sources/ObservationMacros/ObservableMacro.swift
@@ -118,6 +118,7 @@ extension DeclModifierListSyntax {
         case .fileprivate: fallthrough
         case .private: fallthrough
         case .internal: fallthrough
+        case .package: fallthrough
         case .public:
           return false
         default:

--- a/test/ModuleInterface/Observable.swift
+++ b/test/ModuleInterface/Observable.swift
@@ -1,6 +1,6 @@
 // RUN: %empty-directory(%t)
-// RUN: %target-swift-emit-module-interface(%t/Library.swiftinterface) %s -module-name Library -plugin-path %swift-plugin-dir -disable-availability-checking
-// RUN: %target-swift-typecheck-module-from-interface(%t/Library.swiftinterface) -module-name Library -disable-availability-checking
+// RUN: %target-swift-emit-module-interface(%t/Library.swiftinterface) %s -package-name Library -module-name Library -plugin-path %swift-plugin-dir -disable-availability-checking
+// RUN: %target-swift-typecheck-module-from-interface(%t/Library.swiftinterface) -package-name Library -module-name Library -disable-availability-checking
 // RUN: %FileCheck %s < %t/Library.swiftinterface
 
 // REQUIRES: swift_swift_parser
@@ -15,6 +15,7 @@ import Observation
 @Observable
 public class SomeClass {
   public var field = 3
+  package var test = 4
   @ObservationIgnored public var ignored = 4
 }
 

